### PR TITLE
clean up main.tf

### PR DESCRIPTION
**Pull request by duttpathak**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -80,9 +80,8 @@ resource "aws_security_group" "instance" {
      # This IP address range 0.0.0.0/0 includes all. 
     # This allows incoming requests on port 8080 from any IP.
     cidr_blocks = ["0.0.0.0/0"]
-    
+
```
Comment (tarpanpathak) [inline main.tf]: Remove the empty line

File: main.tf [hunk 2]
```diff
@@ -91,25 +90,28 @@ resource "aws_security_group" "instance" {
 resource "aws_instance" "example" {
   ami                    = "ami-0889a44b331db0194"
   instance_type          = "t2.micro"
+  # List of security group IDs to associate with. 
   vpc_security_group_ids = [aws_security_group.instance.id]
-
+  # User data to provide when launching the instance. 
+  # Updates to this filed will trigger a start/stop of
```
Comment (tarpanpathak) [inline main.tf]: **Field** is misspelled. Also, the `user data` itself does not trigger a restart so remove this comment and move it to the appropriate section in the file.

File: main.tf [hunk 3]
```diff
@@ -91,25 +90,28 @@ resource "aws_security_group" "instance" {
...
+  # User data to provide when launching the instance. 
+  # Updates to this filed will trigger a start/stop of 
+  # EC2 instance by default. 
   user_data = <<-EOF
               #!/bin/bash
               echo "Hello, World" > index.html
               nohup busybox httpd -f -p 8080 &
               EOF
-
+  #
```
Comment (tarpanpathak) [inline main.tf]: Add the proper comment here.

File: main.tf [hunk 4]
```diff
@@ -91,25 +90,28 @@ resource "aws_security_group" "instance" {
...
   user_data_replace_on_change = true
 
   tags = {
     Name = "terraform-example"
   }
 }
 
-# Line 93 is used to create an index.html file and with 
+# Line 97 is used to create an index.html file and with
```
Comment (tarpanpathak) [inline main.tf]: Move the comment to the appropriate line.

File: main.tf [hunk 5]
```diff
@@ -91,25 +90,28 @@ resource "aws_security_group" "instance" {
...
     Name = "terraform-example"
   }
 }
 
-# Line 93 is used to create an index.html file and with 
+# Line 97 is used to create an index.html file and with 
 # Hello, World in it. 
 
-# Line 94 is using nohup command which is used to run the script 
+# Line 98 is using nohup command which is used to run the script
```
Comment (tarpanpathak) [inline main.tf]: Same as https://github.com/duttpathak/terraform_learn/pull/1/files#r1199640178

File: main.tf (hunk 4 above)
Comment (duttpathak) [reply main.tf]: When I move the lines it changes the color if when I put #.

Comment (duttpathak) [reply main.tf]: I moved it on top.

File: main.tf [hunk 6]
```diff
@@ -80,37 +80,40 @@ resource "aws_security_group" "instance" {
...
 resource "aws_instance" "example" {
   ami                    = "ami-0889a44b331db0194"
   instance_type          = "t2.micro"
+  # List of security group IDs to associate with. 
   vpc_security_group_ids = [aws_security_group.instance.id]
-
+  # User data to provide when launching the instance. 
+  # Updates to this field will trigger a start/stop of 
+  # EC2 instance by default. 
+# Line 103 used to create an index.html file and with
```
Comment (tarpanpathak) [inline main.tf]: Move this comment above line 103 so it's easier to read/understand

File: main.tf [hunk 7]
```diff
@@ -80,37 +80,40 @@ resource "aws_security_group" "instance" {
...
   ami                    = "ami-0889a44b331db0194"
   instance_type          = "t2.micro"
+  # List of security group IDs to associate with. 
   vpc_security_group_ids = [aws_security_group.instance.id]
-
+  # User data to provide when launching the instance. 
+  # Updates to this field will trigger a start/stop of 
+  # EC2 instance by default. 
+# Line 103 used to create an index.html file and with 
+# Hello, World in it.
```
Comment (tarpanpathak) [inline main.tf]: Same as https://github.com/duttpathak/terraform_learn/pull/1/files#r1199642869

File: main.tf [hunk 8]
```diff
@@ -80,37 +80,40 @@ resource "aws_security_group" "instance" {
...
   instance_type          = "t2.micro"
+  # List of security group IDs to associate with. 
   vpc_security_group_ids = [aws_security_group.instance.id]
-
+  # User data to provide when launching the instance. 
+  # Updates to this field will trigger a start/stop of 
+  # EC2 instance by default. 
+# Line 103 used to create an index.html file and with 
+# Hello, World in it. 
+# Line 104  used to run the script even after you log off.
```
Comment (tarpanpathak) [inline main.tf]: Move this comment above line 104 so it's easier to read/understand

File: main.tf [hunk 9]
```diff
@@ -80,37 +80,40 @@ resource "aws_security_group" "instance" {
...
+  # List of security group IDs to associate with. 
   vpc_security_group_ids = [aws_security_group.instance.id]
-
+  # User data to provide when launching the instance. 
+  # Updates to this field will trigger a start/stop of 
+  # EC2 instance by default. 
+# Line 103 used to create an index.html file and with 
+# Hello, World in it. 
+# Line 104  used to run the script even after you log off. 
+# It continues to run until it is finished.
```
Comment (tarpanpathak) [inline main.tf]: Same as https://github.com/duttpathak/terraform_learn/pull/1/files#r1199642956

File: main.tf [hunk 10]
```diff
@@ -80,37 +80,40 @@ resource "aws_security_group" "instance" {
...
+# Hello, World in it. 
+# Line 104  used to run the script even after you log off. 
+# It continues to run until it is finished. 
   user_data = <<-EOF
               #!/bin/bash
               echo "Hello, World" > index.html
               nohup busybox httpd -f -p 8080 &
               EOF
-
+  # used to terminate any previous pushes to aws server.
```
Comment (tarpanpathak) [inline main.tf]: The comment from lines 95/96 should go here.

**Pull request merged**